### PR TITLE
add gcp jobs for extended registry, builds and image ecosystem

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@smarterclayton @stevekuznetsov ptal.  Note that this makes these tests blocking for all PRs currently.

If we change them to only run for certain paths (like the old jobs did) then we need to start running them nightly also or in some other way so we know if someone/something has broken them.

**Comment by smarterclayton** [comment]
I would say add the run_changed.

Do you need the nightly to pin down when it regressed?  It's useful, but these tests run pretty often (usually every day or two) and there aren't *that* many merges

**Comment by stevekuznetsov** [comment]
CLAYTON THIS ADDS NEW BLOCKING JOBS SO IF YOU MERGE IT MAKE SURE TO TRIGGER THE JOBS FOR ALL PRS OR YOU BLOCK THE MERGE QUEUE

**Comment by bparees** [comment]
> Do  you need the nightly to pin down when it regressed?  It's useful, but  these tests run pretty often (usually every day or two) and there aren't  that many merges

they don't run that frequently in my experience because we've had them break(usually due to upstream image changes) and no one came screaming to us about it.

**Comment by bparees** [comment]
updated to be run_if_changed jobs.  Also removed the aws jobs.

**Comment by bparees** [comment]
@openshift/sig-developer-experience fyi.  changes to our extended job names as they're run against origin.

File: ci-operator/jobs/openshift/origin/openshift-origin-master-presubmits.yaml [hunk 1]
```diff
@@ -1,39 +1,5 @@
 presubmits:
   openshift/origin:
-  - agent: kubernetes
```
Comment (bparees) [inline ci-operator/jobs/openshift/origin/openshift-origin-master-presubmits.yaml]: the generator is removing this. i don't know why.

**Comment by bparees** [comment]
@smarterclayton @stevekuznetsov ok i think this is cleanly generated now.

**Comment by bparees** [comment]
(this also puts us on the cusp of eliminating our nightly jobs because i'm tired of cutting across the grain trying to maintain them)

**Comment by smarterclayton** [comment]
I can run migrate status and remove your old ones after this merges.

File: ci-operator/config/openshift/origin/openshift-origin-master.yaml [hunk 2]
```diff
@@ -202,3 +202,15 @@ tests:
     run-tests
   openshift_ansible:
     cluster_profile: gcp-crio
+- as: extended-builds
```
Comment (smarterclayton) [inline ci-operator/config/openshift/origin/openshift-origin-master.yaml]: call this `e2e-gcp-builds`

File: ci-operator/config/openshift/origin/openshift-origin-master.yaml [hunk 3]
```diff
@@ -202,3 +202,15 @@ tests:
...
+  commands: TEST_FOCUS='\[Feature:Builds\]' TEST_PARALLELISM=5 run-tests
+  openshift_ansible:
+    cluster_profile: gcp
+- as: extended-image-ecosystem
```
Comment (smarterclayton) [inline ci-operator/config/openshift/origin/openshift-origin-master.yaml]: call this `e2e-gcp-image-ecosystem`

File: ci-operator/config/openshift/origin/openshift-origin-master.yaml [hunk 4]
```diff
@@ -202,3 +202,15 @@ tests:
...
+  commands: TEST_FOCUS='\[image_ecosystem\]' run-tests
+  openshift_ansible:
+    cluster_profile: gcp
+- as: extended-image-registry
```
Comment (smarterclayton) [inline ci-operator/config/openshift/origin/openshift-origin-master.yaml]: call this `e2e-gcp-image-registry`

**Comment by bparees** [comment]
> I can run migrate status and remove your old ones after this merges.

are you saying you don't want me to remove the old jobs in this PR?

**Comment by bparees** [comment]
updated names, awaiting confirmation on whether to delete the old jobs or not (currently this PR deletes them)

**Comment by smarterclayton** [comment]
I do want you to.  I just want to get a chance to run migrate status.

**Comment by bparees** [comment]
> I do want you to. I just want to get a chance to run migrate status.

ok, well they are deleted in this PR (i mean i'll have to go remove the aos-cd-jobs too at some point of course) so i think we're good to go?

**Pull request merged**